netbox/netbox_inventory.py

```diff
@@ -132,53 +182,84 @@ def fetch_inventory(cli_opts):
     if cli_opts.region:
         params["region"] = cli_opts.region
 
-    res = netbox.get("/api/dcim/devices/", params=params)
+    res = NETBOX_SESSION.get("/api/dcim/devices/", params=params)
     if not res.ok:
         sys.exit("FAIL: get inventory: " + res.text)
```

Comment:
**Suggestion:** Instead of using `sys.exit` for error handling, raise an exception to allow for more flexible error management and potential recovery from calling functions. [best practice, importance: 7]
```suggestion
        raise Exception("FAIL: get inventory: " + res.text)
```